data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/AbstractDataBackedRestAPIIntegrationTest.java

```diff
@@ -125,15 +128,27 @@
 
   protected ForkChoice forkChoice;
 
-  private void setupStorage(final StateStorageMode storageMode, final boolean useMockForkChoice) {
-    setupStorage(InMemoryStorageSystemBuilder.buildDefault(storageMode), useMockForkChoice);
+  private void setupStorage(
+      final StateStorageMode storageMode,
+      final boolean useMockForkChoice,
+      final SpecMilestone specMilestone) {
+    setupStorage(
+        InMemoryStorageSystemBuilder.buildDefault(storageMode), useMockForkChoice, specMilestone);
   }
 
-  private void setupStorage(final StorageSystem storageSystem, final boolean useMockForkChoice) {
+  private void setupStorage(
+      final StorageSystem storageSystem,
+      final boolean useMockForkChoice,
+      final SpecMilestone specMilestone) {
+    this.spec =
+        specMilestone == SpecMilestone.PHASE0
+            ? TestSpecFactory.createMinimalPhase0()
+            : TestSpecFactory.createMinimalAltair();
```

Comment:
I wonder if we should push this into `TestSpecFactory` to make it easy for other tests to switch specs like this.